Doc/c-api/exceptions.rst

```diff
@@ -402,51 +402,44 @@ Querying the error indicator
 
 .. c:function:: PyObject *PyErr_GetRaisedException(void)
 
-   Returns the exception currently being raised, clearing the exception at
-   the same time. Do not confuse this with the exception currently being
-   handled which can be accessed with  :c:func:`PyErr_GetHandledException`.
+   Return the exception currently being raised, clearing the error indicator at
+   the same time.
 
-   .. note::
+   This function is used by code that needs to catch exceptions,
+   or code that needs to save and restore the error indicator temporarily.
 
-      This function is normally only used by code that needs to catch exceptions or
-      by code that needs to save and restore the error indicator temporarily, e.g.::
+   For example::
 
-         {
-            PyObject *exc = PyErr_GetRaisedException();
+      {
+         PyObject *exc = PyErr_GetRaisedException();
 
-            /* ... code that might produce other errors ... */
+         /* ... code that might produce other errors ... */
 
-            PyErr_SetRaisedException(exc);
-         }
+         PyErr_SetRaisedException(exc);
+      }
 
-   .. versionadded:: 3.12
+   .. seealso::
 
+      * :c:func:`PyErr_SetRaisedException`,
+        to set the current raised exception.
 
-.. c:function:: void PyErr_SetRaisedException(PyObject *exc)
+      * :c:func:`PyErr_GetHandledException`,
+         to fetch the exception currently being handled.
 
-   Sets the exception currently being raised ``exc``.
-   If the exception is already set, it is cleared first.
+   .. versionadded:: 3.12
 
-   ``exc`` must be a valid exception.
-   (Violating this rules will cause subtle problems later.)
-   This call consumes a reference to the ``exc`` object: you must own a
-   reference to that object before the call and after the call you no longer own
-   that reference.
-   (If you don't understand this, don't use this function. I warned you.)
 
-   .. note::
+.. c:function:: void PyErr_SetRaisedException(PyObject *exc)
 
-      This function is normally only used by code that needs to save and restore the
-      error indicator temporarily.  Use :c:func:`PyErr_GetRaisedException` to save
-      the current exception, e.g.::
+   Set *exc* as the exception currently being raised.
+   If another exception is already raised, it is cleared first.
 
-         {
-            PyObject *exc = PyErr_GetRaisedException();
+   .. caution::
```

Comment:
Unfortunately, the existing Python docs theme appears to be completely missing formatting for the caution admonition—in fact, it appears to lack such for every admonition I tested except for `warning` and `note`. So, it seems we have to either have to upgrade this to a warning:

```suggestion
   .. warning::
```

or, downgrade it back to a note:

```suggestion
   .. note::
```

Reply:
I'd prefer a warning. APIs that steal references are a source of bugs; IMO it is worth it to place emphasis on such deviances.